File: factorRoot.java
import java.util.Scanner;
public class factorRoot {
    
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        System.out.println("Insira radicando a ser simplificado:");
        int teste = sc.nextInt();
        simplifyRoot(teste);
        sc.close();
        
    }

    private static void simplifyRoot(int value) 
    {
        int factor = 2;
        int simplifiedRadicand = value;
        int root = 1;
        while (factor <= simplifiedRadicand)
        {
            if (simplifiedRadicand % (factor*factor) == 0)
            {
                simplifiedRadicand /= (factor * factor);
                root *= factor;            
            } 
            else factor++;
        }
        System.out.println("Raiz:" + root);
        if (simplifiedRadicand!=1) {System.out.println("Radicando:" + simplifiedRadicand);} 
    }
}

File: factorRoot.py
import copy


def simplify_root(value):
    factor = 2
    simplified_radicand = copy.deepcopy(value)
    root = 1
    while factor <= simplified_radicand:
        if simplified_radicand % (factor*factor) == 0:
            simplified_radicand /= (factor * factor)
            root *= factor
        else:
            factor = factor + 1
    msg = "A raiz de " + str(value)
    if simplified_radicand != 1:
        msg = msg + " simplificada"
    msg = msg + " é " + str(root)
    if simplified_radicand != 1:
        msg = msg + " √" + str(int(simplified_radicand))
    print(msg)


teste = int(input("Insira radicando a ser simplificado:"))
simplify_root(teste)
